vnext/Microsoft.ReactNative.Cxx/JSValueReader-WinRT.h

```diff
@@ -0,0 +1,203 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include <CppWinRTIncludes.h>
+#include <UI.Xaml.Markup.h>
+#include <chrono>
+
+#include "JSValue.h"
+#include "JSValueTreeReader.h"
+#include "StructInfo.h"
+
+#include "winrt/Microsoft.ReactNative.h"
+
+#include <string>
+
+namespace winrt::Microsoft::ReactNative {
+inline BYTE GetAFromArgb(DWORD v) {
+  return ((BYTE)((v & 0xFF000000) >> 24));
+}
+inline BYTE GetRFromArgb(DWORD v) {
+  return ((BYTE)((v & 0x00FF0000) >> 16));
+}
+inline BYTE GetGFromArgb(DWORD v) {
+  return ((BYTE)((v & 0x0000FF00) >> 8));
+}
+inline BYTE GetBFromArgb(DWORD v) {
+  return ((BYTE)((v & 0x000000FF)));
+}
+
+inline ui::Color ColorFromArgb(const JSValue &value) {
+  auto argb = static_cast<DWORD>(value.AsInt64());
+  return winrt::ColorHelper::FromArgb(GetAFromArgb(argb), GetRFromArgb(argb), GetGFromArgb(argb), GetBFromArgb(argb));
+}
+
+inline ui::Color UndefinedColor() noexcept {
+  static auto c = ColorHelper::FromArgb(1, 0, 0, 0);
+  return c;
+}
+
+inline xaml::Media::SolidColorBrush UndefinedSolidColorBrush() noexcept {
+  static auto b = xaml::Media::SolidColorBrush(UndefinedColor());
+  return b;
+}
+
+inline void ReadValue(IJSValueReader const &reader, /*out*/ ui::Color &value) {
+  value = UndefinedColor();
+  if (reader.ValueType() == JSValueType::Int64) {
+    auto argb = static_cast<DWORD>(reader.GetInt64());
+    value =
+        winrt::ColorHelper::FromArgb(GetAFromArgb(argb), GetRFromArgb(argb), GetGFromArgb(argb), GetBFromArgb(argb));
+  }
+}
+
+struct ColorComp {
+  bool operator()(const ui::Color &lhs, const ui::Color &rhs) const {
+    return (
```

Comment:
Potential optimization here if color is just a packed 32 bit struct is:

1. static assert color struct 32 bit
2. Reinterpret cast struct to 32 bit uint and compare